### PR TITLE
JAMES-3264: MAILBOX details are read 3 times upon indexing

### PR DESCRIPTION
**Pull request by nvduc91**

### CODE REVIEW

File: mailbox/store/src/main/java/org/apache/james/mailbox/store/search/ListeningMessageSearchIndex.java [hunk 1]
```diff
@@ -84,13 +84,13 @@ public boolean isHandling(Event event) {
 
             return factory.getMailboxMapper(session)
                 .findMailboxById(mailboxId)
-                .flatMap(mailbox -> delete(session, mailbox, expunged.getUids()));
+                .flatMap(mailbox -> delete(session, mailbox.getMailboxId(), expunged.getUids()));
```
Comment (chibenwa) [inline mailbox/store/src/main/java/org/apache/james/mailbox/store/search/ListeningMessageSearchIndex.java:87]: Maybe we can avoid reading this mailbox the line above?

File: mailbox/store/src/main/java/org/apache/james/mailbox/store/search/ListeningMessageSearchIndex.java [hunk 2]
```diff
@@ -84,13 +84,13 @@ public boolean isHandling(Event event) {
...
         } else if (event instanceof FlagsUpdated) {
             FlagsUpdated flagsUpdated = (FlagsUpdated) event;
 
             return factory.getMailboxMapper(session)
                 .findMailboxById(mailboxId)
-                .flatMap(mailbox -> update(session, mailbox, flagsUpdated.getUpdatedFlags()));
+                .flatMap(mailbox -> update(session, mailbox.getMailboxId(), flagsUpdated.getUpdatedFlags()));
```
Comment (chibenwa) [inline mailbox/store/src/main/java/org/apache/james/mailbox/store/search/ListeningMessageSearchIndex.java:93]: Maybe we can avoid reading this mailbox the line above?

File: mailbox/store/src/main/java/org/apache/james/mailbox/store/search/ListeningMessageSearchIndex.java [hunk 3]
```diff
@@ -123,10 +123,10 @@ public boolean isHandling(Event event) {
      * Delete the concerned UIDs for the given {@link Mailbox} from the index
      *
      * @param session      The mailbox session performing the expunge
-     * @param mailbox      mailbox on which the expunge was performed
+     * @param mailboxId      mailbox on which the expunge was performed
```
Comment (chibenwa) [inline mailbox/store/src/main/java/org/apache/james/mailbox/store/search/ListeningMessageSearchIndex.java:126]: mailboxId on which the expunge was performed

File: mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/ElasticSearchWithCassandraMailboxTest.java [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+import org.apache.james.mailbox.store.mail.MessageMapper.FetchType;
+import org.apache.james.metrics.tests.RecordingMetricFactory;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import com.datastax.driver.core.BoundStatement;
+import com.google.common.collect.ImmutableSet;
+
+public class ElasticSearchWithCassandraMailboxTest {
```
Comment (chibenwa) [inline mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/ElasticSearchWithCassandraMailboxTest.java:74]: This test is not conclusive, maybe we should not merge it?

Comment (nvduc91) [reply mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/ElasticSearchWithCassandraMailboxTest.java:74]: Yes, that why i put it in draft

Comment (chibenwa) [reply mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/ElasticSearchWithCassandraMailboxTest.java:74]: IMO let's remove it, unless you write some tests for the changeset you introduced (update / delete)

Comment (remk) [reply mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/ElasticSearchWithCassandraMailboxTest.java:74]: Could be kept as some sort of a non regression performance test ?
But i am not sure of the value of it

File: mailbox/cassandra/pom.xml [hunk 5]
```diff
@@ -78,6 +95,17 @@
...
+            <groupId>${james.groupId}</groupId>
+            <artifactId>james-server-util</artifactId>
+            <type>test-jar</type>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>${james.groupId}</groupId>
+            <artifactId>james-server-testing</artifactId>
+            <scope>test</scope>
+        </dependency>
```
Comment (chibenwa) [inline mailbox/cassandra/pom.xml:108]: Can we then remove everything added in that POM ?

**Comment by Arsnael** [comment]
```
[698239ba34a09c931dae92b66fbc8221b519a1af] [INFO] There is 1 error reported by Checkstyle 8.29 with checkstyle.xml ruleset.
[698239ba34a09c931dae92b66fbc8221b519a1af] [ERROR] src/main/java/org/apache/james/mailbox/elasticsearch/events/ElasticSearchListeningMessageSearchIndex.java:[73,8] (imports) UnusedImports: Unused import - com.github.steveash.guavate.Guavate.
```

**Comment by nvduc91** [comment]
```
/james-project/mailbox/tools/indexer/src/main/java/org/apache/mailbox/tools/indexer/ReIndexerPerformer.java:[335,85] incompatible types: org.apache.james.mailbox.model.Mailbox cannot be converted to org.apache.james.mailbox.model.MailboxId
```

**Comment by nvduc91** [comment]
rebase master to fix

**Comment by nvduc91** [comment]
test this please 

```
[2fdd89acef6dbd8c131bcc6b42263a7352236b49] [ERROR] Failures: 
[2fdd89acef6dbd8c131bcc6b42263a7352236b49] [ERROR]   ElasticSearchIntegrationTest>AbstractMessageSearchIndexTest.searchShouldReturnSeenMessagesWhenFlagsGotUpdated:546 
[2fdd89acef6dbd8c131bcc6b42263a7352236b49] Expecting:
[2fdd89acef6dbd8c131bcc6b42263a7352236b49]  <[MessageUid{uid=6}]>
[2fdd89acef6dbd8c131bcc6b42263a7352236b49] to contain:
[2fdd89acef6dbd8c131bcc6b42263a7352236b49]  <[MessageUid{uid=5}]>
[2fdd89acef6dbd8c131bcc6b42263a7352236b49] but could not find:
[2fdd89acef6dbd8c131bcc6b42263a7352236b49]  <[MessageUid{uid=5}]>
```

**Comment by chibenwa** [comment]
I think it is related

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**